### ISSUE TITLE
errors when loading model trained by myself for evaluation

### ISSUE THREAD

**Issue by kailigo** [huanghoujing/person-reid-triplet-loss-baseline#3]
I can use the provided model for evaluation and get reasonable results, but meet errors when loading the model trained by myself for evaluation. The following is the script I used to test using my own trained model:

> python script/experiment/train.py \
> -d '(0,)' \
> --only_test true \
> --dataset duke \
> --last_conv_stride 1 \
> --normalize_feature false \
> --exp_dir /export/reid_datasets/transformed_collection/trained_models/duke_stride1_evaluation \
> --model_weight_file /export/reid_datasets/transformed_collection/self_train/market_weight_c1000_t1_x0/ckpt.pth

The below is the error log: 

> Keys not found in source state_dict: 
> 	 base.layer2.1.conv3.weight
> 	 base.layer3.0.conv2.weight
> 	 base.layer3.2.bn2.bias
> 	 base.layer1.0.bn2.running_mean
> 	 base.layer4.0.conv2.weight
> 	 base.layer2.0.conv2.weight
> 	 base.layer4.1.bn3.running_mean
> 	 base.layer1.2.bn1.weight
> 	 base.layer2.3.bn3.bias
> 	 base.layer3.5.bn3.bias
> 	 base.layer1.0.bn3.weight
> 	 base.layer2.1.bn3.weight
> 	 base.layer3.0.bn3.running_mean
> 	 base.layer3.5.conv2.weight
> 	 base.layer1.0.bn1.running_var
> 	 base.bn1.running_mean
> 	 base.layer3.3.bn1.weight
> 	 base.layer3.5.conv1.weight
> 	 base.layer4.2.bn3.running_var
> 	 base.layer3.2.bn1.running_var
> 	 base.layer3.0.bn3.running_var
> 	 base.layer3.2.bn3.bias
> 	 base.layer3.5.bn2.weight
> 	 base.layer2.2.bn1.running_var
> 	 base.layer1.1.bn1.weight
> 	 base.layer2.2.bn2.weight
> 	 base.layer2.2.bn1.running_mean
> 	 base.layer1.0.conv3.weight
> 	 base.layer3.2.conv2.weight
> 	 base.layer4.1.bn3.running_var
> 	 base.layer3.0.downsample.1.running_var
> 	 base.layer3.2.bn3.running_var
> 	 base.layer2.2.bn2.running_var
> 	 base.layer2.0.bn3.running_var
> 	 base.layer3.2.bn1.running_mean
> 	 base.layer2.2.bn3.running_mean
> 	 base.layer3.1.bn1.running_mean
> 	 base.layer4.2.bn2.weight
> 	 base.layer3.2.bn2.running_mean
> 	 base.layer4.0.conv1.weight
> 	 base.layer3.3.bn3.bias
> 	 base.layer2.3.conv2.weight
> 	 base.layer4.1.bn2.running_mean
> 	 base.layer4.0.downsample.0.weight
> 	 base.layer4.1.conv3.weight
> 	 base.layer3.3.conv2.weight
> 	 base.layer3.4.conv2.weight
> 	 base.layer4.2.conv1.weight
> 	 base.layer3.5.bn1.bias
> 	 base.layer3.5.bn3.running_var
> 	 base.layer1.2.bn2.running_mean
> 	 base.layer1.1.conv2.weight
> 	 base.layer1.1.bn3.bias
> 	 base.layer1.0.conv1.weight
> 	 base.layer2.2.conv2.weight
> 	 base.layer2.1.bn1.running_mean
> 	 base.layer1.1.bn3.running_var
> 	 base.layer1.2.bn3.running_mean
> 	 base.layer3.1.bn3.running_var
> 	 base.layer2.2.bn2.bias
> 	 base.layer2.1.bn2.weight
> 	 base.layer3.2.bn2.weight
> 	 base.layer2.0.downsample.1.weight
> 	 base.layer3.0.downsample.1.weight
> 	 base.layer3.3.bn2.running_mean
> 	 base.layer2.3.bn2.bias
> 	 base.layer4.0.bn2.weight
> 	 base.layer3.0.bn2.weight
> 	 base.layer3.2.bn3.weight
> 	 base.layer2.1.bn2.running_mean
> 	 base.layer2.1.bn3.bias
> 	 base.layer2.3.bn1.running_mean
> 	 base.layer4.0.bn2.running_var
> 	 base.layer1.1.conv1.weight
> 	 base.layer3.3.conv1.weight
> 	 base.layer3.5.bn1.running_var
> 	 base.layer1.2.bn2.running_var
> 	 base.layer3.4.bn3.running_mean
> 	 base.layer1.1.conv3.weight
> 	 base.layer1.1.bn3.running_mean
> 	 base.layer1.0.bn2.running_var
> 	 base.layer3.3.bn1.running_var
> 	 base.layer4.0.bn1.running_var
> 	 base.layer1.0.downsample.1.weight
> 	 base.layer3.5.conv3.weight
> 	 base.layer1.0.downsample.1.running_var
> 	 base.layer1.0.bn3.bias
> 	 base.layer3.4.bn1.running_var
> 	 base.layer3.4.bn2.bias
> 	 base.layer2.3.bn2.running_var
> 	 base.layer3.3.bn3.running_var
> 	 base.layer3.4.conv3.weight
> 	 base.layer3.5.bn3.weight
> 	 base.layer3.1.bn2.running_mean
> 	 base.layer2.2.bn3.running_var
> 	 base.layer3.5.bn2.bias
> 	 base.layer2.2.conv1.weight
> 	 base.layer4.0.bn1.bias
> 	 base.layer1.2.bn1.running_var
> 	 base.layer4.2.bn2.running_var
> 	 base.layer4.2.conv3.weight
> 	 base.layer4.2.bn2.bias
> 	 base.layer3.4.bn1.bias
> 	 base.bn1.bias
> 	 base.layer2.0.bn2.bias
> 	 base.layer3.3.bn1.bias
> 	 base.layer1.2.bn1.running_mean
> 	 base.layer4.0.bn3.running_mean
> 	 base.layer2.3.bn1.weight
> 	 base.layer2.0.bn2.weight
> 	 base.layer3.2.bn3.running_mean
> 	 base.layer3.0.bn2.running_var
> 	 base.layer3.2.bn2.running_var
> 	 base.layer2.3.bn3.running_mean
> 	 base.layer1.1.bn1.running_var
> 	 base.layer1.0.bn1.bias
> 	 base.layer1.2.conv1.weight
> 	 base.layer4.1.bn1.running_mean
> 	 base.layer1.2.bn3.running_var
> 	 base.layer1.2.bn2.bias
> 	 base.layer1.1.bn2.running_var
> 	 base.layer4.0.bn1.weight
> 	 base.layer2.0.downsample.0.weight
> 	 base.layer4.0.bn2.running_mean
> 	 base.layer2.2.bn3.bias
> 	 base.layer3.5.bn2.running_var
> 	 base.layer4.0.bn3.running_var
> 	 base.layer2.1.bn2.bias
> 	 base.layer3.0.bn2.bias
> 	 base.layer3.0.bn2.running_mean
> 	 base.layer3.4.bn1.weight
> 	 base.layer2.0.downsample.1.running_mean
> 	 base.conv1.weight
> 	 base.layer4.1.bn2.running_var
> 	 base.layer1.0.downsample.1.running_mean
> 	 base.layer4.1.bn1.weight
> 	 base.layer2.0.bn2.running_mean
> 	 base.layer3.0.downsample.0.weight
> 	 base.layer4.1.bn1.bias
> 	 base.layer2.2.bn1.bias
> 	 base.layer1.2.conv3.weight
> 	 base.layer1.0.downsample.1.bias
> 	 base.layer4.0.bn3.weight
> 	 base.layer2.2.bn1.weight
> 	 base.layer3.2.conv1.weight
> 	 base.layer3.0.bn1.bias
> 	 base.layer2.2.conv3.weight
> 	 base.layer4.1.bn2.weight
> 	 base.layer1.0.bn2.weight
> 	 base.layer3.0.bn3.weight
> 	 base.layer3.5.bn1.running_mean
> 	 base.layer2.3.bn1.running_var
> 	 base.layer3.0.bn1.weight
> 	 base.layer4.0.downsample.1.running_mean
> 	 base.layer1.1.bn2.bias
> 	 base.layer3.5.bn1.weight
> 	 base.layer3.0.bn3.bias
> 	 base.layer3.2.conv3.weight
> 	 base.layer3.1.bn3.bias
> 	 base.layer4.0.conv3.weight
> 	 base.layer3.4.bn3.running_var
> 	 base.layer2.0.bn3.weight
> 	 base.layer2.1.bn1.bias
> 	 base.layer3.0.conv1.weight
> 	 base.layer2.3.bn3.running_var
> 	 base.layer3.4.bn2.running_mean
> 	 base.layer3.3.bn3.weight
> 	 base.layer2.1.bn3.running_mean
> 	 base.layer3.5.bn3.running_mean
> 	 base.layer3.4.bn3.bias
> 	 base.layer3.1.bn1.weight
> 	 base.layer4.0.bn3.bias
> 	 base.layer4.1.bn3.weight
> 	 base.layer3.3.conv3.weight
> 	 base.layer4.2.bn1.running_var
> 	 base.layer4.2.bn3.weight
> 	 base.layer2.3.bn1.bias
> 	 base.layer3.2.bn1.weight
> 	 base.layer1.0.bn1.running_mean
> 	 base.layer2.0.conv1.weight
> 	 base.layer2.0.bn1.weight
> 	 base.layer3.1.bn3.weight
> 	 base.layer3.1.bn1.running_var
> 	 base.layer3.4.bn2.running_var
> 	 base.layer2.0.bn3.running_mean
> 	 base.layer4.2.bn3.bias
> 	 base.layer3.3.bn1.running_mean
> 	 base.layer4.2.bn1.weight
> 	 base.bn1.weight
> 	 base.layer1.1.bn2.weight
> 	 base.layer1.0.conv2.weight
> 	 base.layer2.0.bn1.bias
> 	 base.layer1.2.bn2.weight
> 	 base.layer1.1.bn1.bias
> 	 base.layer3.4.bn2.weight
> 	 base.layer4.2.bn3.running_mean
> 	 base.layer2.0.conv3.weight
> 	 base.layer2.2.bn3.weight
> 	 base.layer3.5.bn2.running_mean
> 	 base.layer3.0.downsample.1.bias
> 	 base.layer1.2.bn3.weight
> 	 base.layer1.2.conv2.weight
> 	 base.layer2.0.bn2.running_var
> 	 base.layer2.1.bn3.running_var
> 	 base.layer3.0.bn1.running_mean
> 	 base.layer2.0.bn3.bias
> 	 base.layer3.0.bn1.running_var
> 	 base.layer4.2.bn2.running_mean
> 	 base.layer1.2.bn3.bias
> 	 base.layer4.2.conv2.weight
> 	 base.layer1.0.bn1.weight
> 	 base.layer2.3.bn3.weight
> 	 base.layer3.1.bn2.weight
> 	 base.layer4.1.bn2.bias
> 	 base.layer2.0.bn1.running_var
> 	 base.layer2.0.downsample.1.running_var
> 	 base.layer2.3.bn2.weight
> 	 base.layer2.3.conv3.weight
> 	 base.layer3.1.bn1.bias
> 	 base.layer1.0.downsample.0.weight
> 	 base.layer2.3.bn2.running_mean
> 	 base.layer4.0.bn2.bias
> 	 base.layer3.3.bn2.bias
> 	 base.layer2.1.conv2.weight
> 	 base.layer3.1.conv3.weight
> 	 base.layer3.1.bn2.bias
> 	 base.layer1.1.bn1.running_mean
> 	 base.layer3.2.bn1.bias
> 	 base.layer3.3.bn3.running_mean
> 	 base.layer1.2.bn1.bias
> 	 base.layer4.0.downsample.1.running_var
> 	 base.layer3.3.bn2.weight
> 	 base.layer4.0.downsample.1.bias
> 	 base.layer2.2.bn2.running_mean
> 	 base.layer3.4.bn1.running_mean
> 	 base.layer4.0.bn1.running_mean
> 	 base.layer4.1.conv2.weight
> 	 base.layer3.4.conv1.weight
> 	 base.layer2.3.conv1.weight
> 	 base.layer1.0.bn3.running_var
> 	 base.layer3.0.conv3.weight
> 	 base.layer3.1.bn2.running_var
> 	 base.layer2.1.bn2.running_var
> 	 base.layer3.1.conv2.weight
> 	 base.layer2.0.downsample.1.bias
> 	 base.layer4.1.bn3.bias
> 	 base.layer1.1.bn3.weight
> 	 base.layer2.1.bn1.running_var
> 	 base.layer1.1.bn2.running_mean
> 	 base.layer1.0.bn3.running_mean
> 	 base.layer3.3.bn2.running_var
> 	 base.layer3.1.bn3.running_mean
> 	 base.layer2.0.bn1.running_mean
> 	 base.layer2.1.conv1.weight
> 	 base.layer1.0.bn2.bias
> 	 base.layer3.0.downsample.1.running_mean
> 	 base.layer4.2.bn1.bias
> 	 base.bn1.running_var
> 	 base.layer4.0.downsample.1.weight
> 	 base.layer3.4.bn3.weight
> 	 base.layer3.1.conv1.weight
> 	 base.layer4.1.bn1.running_var
> 	 base.layer4.2.bn1.running_mean
> 	 base.layer2.1.bn1.weight
> 	 base.layer4.1.conv1.weight
> Keys not found in destination state_dict: 
> 	 ep
> 	 scores
> 	 state_dicts
> 

How does this happens? Thanks!

**Comment by huanghoujing** [comment]
Thank you for your attention.

Here changing `--model_weight_file /export/reid_datasets/transformed_collection/self_train/market_weight_c1000_t1_x0/ckpt.pth` to `--ckpt_file /export/reid_datasets/transformed_collection/self_train/market_weight_c1000_t1_x0/ckpt.pth` should solve your problem.

This is the same problem as [explained here](https://github.com/huanghoujing/beyond-part-models/issues/2).
